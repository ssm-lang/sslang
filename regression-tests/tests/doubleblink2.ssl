toggle(led : &Int) -> () =
  led <- (1 - deref (led: &Int))

slow(led : &Int) -> () =
  let e1 = new ()
  loop
    (toggle: &Int -> ()) (led: &Int)
    after 30 , e1 <- ()
    wait e1

fast(led : &Int) -> () =
  let e2 = (new () : &())
  loop
    (toggle: &Int -> ()) (led: &Int)
    after 20 , e2 <- ()
    wait e2

main(led : &Int) -> () =
  par (slow: &Int -> ()) (led: &Int)
      (fast: &Int -> ()) led
