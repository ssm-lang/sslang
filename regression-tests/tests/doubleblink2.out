          20 led: 1
          30 led: 0
          40 led: 1
          60 led: 1
          80 led: 0
          90 led: 1
         100 led: 0
         120 led: 0
         140 led: 1
         150 led: 0
         160 led: 1
         180 led: 1
         200 led: 0
         210 led: 1
         220 led: 0
         240 led: 0
         260 led: 1
         270 led: 0
         280 led: 1
         300 led: 1
         320 led: 0
         330 led: 1
         340 led: 0
         360 led: 0
         380 led: 1
         390 led: 0
         400 led: 1
         420 led: 1
         440 led: 0
         450 led: 1
         460 led: 0
         480 led: 0
         500 led: 1
